### PR TITLE
Code refactors and created validation service

### PR DESCRIPTION
**Pull request by matiasrb97**
## Description
Please describe the changes.
If it's a feature or enhancement please be as detailed as possible.
If it's a bug fix, please link the issue that it fixes or describe the bug in as much detail.

## Requirements Checklist
- [ ] Unit Tests updated or fixed
- [ ] Docs/guides updated

### CODE REVIEW

File: lib/src/cmaf/ham/services/validate.ts [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+ * @param presentation - Presentation from cmaf ham model
+ *
+ * @returns Validation
+ *
+ * @group CMAF
+ *
+ * @alpha
+ *
+ */
+function validatePresentation(presentation: Presentation): Validation {
```
Comment (NoeliaBentancor) [inline lib/src/cmaf/ham/services/validate.ts:40]: This functions seem to be very similar between them, seems to add a lot of code.

Do you think we can make kind of an abstraction so as to have less code?

In the future, if we would like to add more validations , it seems to be hard to know where to do it. We could separate methods , for instance, one that validate identification and we could use it then.
What do you think about it?

**Review by DuckyCB — APPROVED** [review]

File: lib/src/cmaf/ham/services/validate.ts (hunk 1 above)
Comment (matiasrb97) [reply lib/src/cmaf/ham/services/validate.ts:40]: Not sure about doing an abstraction for just validating the id, also I'm seeing that the message is distinct depending on the type. So, I would prefer to keep this implementation taking into account that we are doing simple validations for now to have a base line for future work. I expect that in the future we will have other files to hold complex methods for deeper validations. 

Is it okay for you?

File: lib/src/cmaf/ham/services/validate.ts [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+ *
+ * @group CMAF
+ *
+ * @alpha
+ *
+ */
+function validateSelectionSets(
+	selectionSets: SelectionSet[],
+	presentationId?: string,
+	prevValidation?: Validation,
```
Comment (DuckyCB) [inline lib/src/cmaf/ham/services/validate.ts:75]: `prevValidation` is not defined on jsocs, this applies to all code occurencies

Comment (matiasrb97) [reply lib/src/cmaf/ham/services/validate.ts:75]: Done!

**Pull request merged**